### PR TITLE
Public search API docs

### PR DESCRIPTION
**Pull request by Nimphal**
For #752

### CODE REVIEW

File: portality/view/api_v1.py [hunk 1]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
+        in: path
         required: true
-        type: "string"
+        type: string
       -
-        name: "search_query"
-        description: "this is definitely a very useful place to put some description of each parameter"
-        in: "path"
+        name: page
+        description: Which page of the results you wish to see. This field is optional.
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: Is it necessary to say this field is optional? There is a required:false below it. If it is necessary (e.g. for visibility) then the pageSize one below, starting at line 64, does *not* have the sentence, but also has required:false declared.

File: portality/view/api_v1.py [hunk 2]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
+        in: path
+        required: false
+        type: integer
+      -
+        name: pageSize
+        description: How many results per page you wish to see, the default is 100.
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: Can we get the default to be a constant defined in this module? So the docs and the actual code use the same value?

File: portality/view/api_v1.py [hunk 3]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
+                    format: dateTime
+            query:
+              type: string
+            total:
+              type: integer
+            page:
+              type: integer
+      400:
+        description: Bad Request
+    """
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: A bit long-ish but good I think. I'm sure we can move off the docstrings to someplace else if really necessary.

File: portality/view/api_v1.py [hunk 4]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
+    # get the values for the 2 other bits of search info: the page number and the page size
+    page = request.values.get("page", 1)
+    psize = request.values.get("pageSize", 10)
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: hm, @Nimphal pageSize seems to be 10? So definitely go for definiing DEFAULT_PAGE_SIZE in the api common.py file or at least top of this module.

File: portality/view/api_v1.py [hunk 5]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
+        in: path
         required: true
-        type: "string"
+        type: string
       -
-        name: "page"
-        description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. In a augue nulla. In vehicula est eget erat iaculis, et lacinia enim faucibus. Aliquam diam dolor, euismod ultrices viverra vitae, congue sit amet massa. Duis vestibulum, quam ut ultrices sagittis, nunc diam cursus odio, sed efficitur eros erat a sapien."
-        in: "body"
+        name: page
+        description: Which page of the results you wish to see. This field is optional.
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: same comments as for the journal route at same place

File: portality/view/api_v1.py [hunk 6]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
-        type: "int"
+        type: integer
       -
-        name: "pageSize"
-        description: "Wooo, we can write small essays in here as well. Again limited as it is a string, but pretty neat"
-        in: "body"
-        required: false,
-        type: "int"
+        name: pageSize
+        description: How many results per page you wish to see, the default is 100.
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: same comments as for the journal route at same place, maybe have a DEFAULT_ARTICLE_PAGE_SIZE and DEFAULT_JOURNAL_PAGE_SIZE if they're different.

File: portality/view/api_v1.py (hunk 1 above)
Comment (Nimphal) [inline portality/view/api_v1.py]: It's not necessary, it was just description filler, can remove if needed.

File: portality/view/api_v1.py [hunk 7]
```diff
@@ -40,61 +40,322 @@ def list_operations():
...
+              type: integer
+            page:
+              type: integer
       400:
         description: "Bad Request"
     """
+
     # get the values for the 2 other bits of search info: the page number and the page size
     page = request.values.get("page", 1)
     psize = request.values.get("pageSize", 10)
```
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: and yeah, page size is 10 for articles too, in the code

**Comment by emanuil-tolev** [comment]
OK to recap for others looking at this you've split the search endpoint into 2 for searching articles and journals, and removed the dispatcher endpoint, so that we can write separate documentation for each, I think owing to some documentation framework limitations. This is fine by me, if that's the right abstraction then ok.

File: portality/view/api_v1.py (hunk 6 above)
Comment (Nimphal) [inline portality/view/api_v1.py]: Tested this, doesn't work with the current framework.

**Comment by emanuil-tolev** [comment]
OK, yeah, the swagger extension evaluates the docstrings before Python gets to expanding them (?!) so putting .format on them doesn't work :/. Ignore those comments @Nimphal

File: portality/view/api_v1.py (hunk 1 above)
Comment (emanuil-tolev) [inline portality/view/api_v1.py]: Yeah, if the UI makes it clear in some other way that this field is optional / not required then no need to have a string that says the same.

**Pull request merged**